src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs

```diff
@@ -0,0 +1,307 @@
+            {
+                yield return CreateT(seed++);
+            }
+        }
+
+        protected PriorityQueue<TElement, TPriority> GenericPriorityQueueFactory(
+            int initialCapacity, int countOfItemsToGenerate, out List<(TElement element, TPriority priority)> generatedItems)
+        {
+            generatedItems = this.GenericIEnumerableFactory(countOfItemsToGenerate).ToList();
+
+            var queue = new PriorityQueue<TElement, TPriority>(initialCapacity);
+            foreach (var (element, priority) in generatedItems)
+            {
+                queue.Enqueue(element, priority);
+            }
+
+            return queue;
+        }
+
+        #endregion
+
+        #region Constructors
+
+        [Fact]
+        public void PriorityQueue_Generic_Constructor()
+        {
+            var queue = new PriorityQueue<TElement, TPriority>();
+
+            Assert.Equal(expected: 0, queue.Count);
+            Assert.Empty(queue.UnorderedItems);
+            Assert.Equal(queue.Comparer, Comparer<TPriority>.Default);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_Constructor_int(int initialCapacity)
+        {
+            var queue = new PriorityQueue<TElement, TPriority>(initialCapacity);
+
+            Assert.Empty(queue.UnorderedItems);
+        }
+
+        [Fact]
+        public void PriorityQueue_Generic_Constructor_int_Negative_ThrowsArgumentOutOfRangeException()
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("initialCapacity", () => new PriorityQueue<TElement, TPriority>(-1));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("initialCapacity", () => new PriorityQueue<TElement, TPriority>(int.MinValue));
+        }
+
+        [Fact]
+        public void PriorityQueue_Generic_Constructor_IComparer()
+        {
+            IComparer<TPriority> comparer = Comparer<TPriority>.Default;
+            var queue = new PriorityQueue<TElement, TPriority>(comparer);
+
+            Assert.Equal(comparer, queue.Comparer);
+        }
+
+        [Fact]
+        public void PriorityQueue_Generic_Constructor_IComparer_Null()
+        {
+            var queue = new PriorityQueue<TElement, TPriority>((IComparer<TPriority>)null);
+            Assert.Equal(Comparer<TPriority>.Default, queue.Comparer);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_Constructor_int_IComparer(int initialCapacity)
+        {
+            IComparer<TPriority> comparer = Comparer<TPriority>.Default;
+            var queue = new PriorityQueue<TElement, TPriority>(initialCapacity);
+
+            Assert.Empty(queue.UnorderedItems);
+            Assert.Equal(comparer, queue.Comparer);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_Constructor_IEnumerable(int count)
```

Comment:
There should be a test that checks for null items to enqueue.

Reply:
Covered in `PriorityQueue_Generic_Enqueue_Null`.